src/wallet/wallet_tx_builder.h

```diff
@@ -403,11 +403,11 @@ class WalletTxBuilder {
     ResolveInputsAndPayments(
             CWallet& wallet,
             const ZTXOSelector& selector,
-            SpendableInputs& spendable,
+            SpendableInputs spendable,
```

Comment:
We specifically need a copy here?

Reply:
I _think_ so. Basically, there are four paths through `PrepareTransaction` – fixed fee vs conventional fee and `vector<Payment>` vs `NetAmountRecipient`. `ResolveInputsAndPayments` is only used in the fixed fee & `vector<Payment>` case, and it’s the only place we modify the provided `SpendableInputs`.

Making this copy means that we can use `const SpendableInputs&` in the rest of the code, which avoids incorrectly signaling that `PrepareTransaction` meaningfully modifies the `SpendableInputs&`, when the correct value is actually gotten from the `TransactionEffects` result.

Although we could instead make this `const SpendableInputs&` and explicitly copy it in the body of the method, before calling `LimitToAmount`, which is what does the modification. Actually, I like that better, as it preserves the `const` across more calls, so I’ll make that change.

Reply:
Ok, I made this change as a separate commit, as I’m not 100% sold on it.